server/src/main/java/org/elasticsearch/gateway/IncrementalClusterStateWriter.java

```diff
@@ -244,10 +229,10 @@ private long writeGlobalState(AtomicClusterStateWriter writer, MetaData newMetaD
     }
 
     // exposed for tests
-    static Set<Index> getRelevantIndices(ClusterState state, ClusterState previousState, Set<Index> previouslyWrittenIndices) {
+    static Set<Index> getRelevantIndices(ClusterState state) {
         Set<Index> relevantIndices;
         if (isDataOnlyNode(state)) {
```

Comment:
Seeing as how we're touching this, if we checked `state.nodes().getLocalNode().isMasterNode()` first then we could say merely `else if (state.nodes().getLocalNode(). isDataOnlyNode())` and ditch the `isDataOnlyNode` method.

And there's no need for the intermediate `relevantIndices`, just return the thing already :)

Reply:
++